### PR TITLE
[BFN] Move qsfp eeprom reading to new cached api

### PR DESCRIPTION
**Pull request by dmytroxIntel**
#### Why I did it
- Moving to new sonic xcvr api requires exposing eeprom pages, by that time platform api was providing only PAGE0_LOWER and PAGE0_UPPER. That led threshold specific tests to fail since sff3686 declares threshold information on the third page.

#### How I did it
- Used new api exposed from platform side in order to make it possible to read all pages.

#### How to verify it
- Run sonic mgmt sfp tests.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py [hunk 1]
```diff
@@ -5,35 +5,45 @@
...
 SFP_TYPE = "SFP"
 QSFP_TYPE = "QSFP"
 QSFP_DD_TYPE = "QSFP_DD"
+SFP_EEPROM_PATH = "/var/run/platform/sfp/"
+EEPROM_PAGE_SIZE = 128
 
+try:
+    from thrift.Thrift import TApplicationException
+
+    def cached_num_bytes_get(client):
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py:21]: Can we move this as a part of `__init__()` ?

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py [hunk 2]
```diff
@@ -5,35 +5,45 @@
...
     from sonic_platform.platform_thrift_client import thrift_try
     from sonic_platform.platform_thrift_client import pltfm_mgr_try
+    from sonic_platform.pltfm_mgr_rpc import ttypes
 except ImportError as e:
     raise ImportError (str(e) + "- required module not found")
 
 SFP_TYPE = "SFP"
 QSFP_TYPE = "QSFP"
 QSFP_DD_TYPE = "QSFP_DD"
+SFP_EEPROM_PATH = "/var/run/platform/sfp/"
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py:15]: Better to keep this as a class variable

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py (hunk 1 above)
Comment (dmytroxIntel) [reply platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py:20]: It's better to keep it here, so it executes once and not every time constructor called

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py (hunk 2 above)
Comment (dmytroxIntel) [reply platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py]: +

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py [hunk 3]
```diff
@@ -84,6 +94,27 @@ def qsfp_info_get(client):
...
+
+        page = None
+
+        if offset // EEPROM_PAGE_SIZE == 0:
+            page = ttypes.qsfp_eeprom_page_t.PAGE0_LOWER
+        elif offset // EEPROM_PAGE_SIZE == 1:
+            page = ttypes.qsfp_eeprom_page_t.PAGE0_UPPER
+        elif offset // EEPROM_PAGE_SIZE == 4:
+            page = ttypes.qsfp_eeprom_page_t.PAGE3
+
```
[the remark this answers is not in the document]
Comment (dmytroxIntel) [reply platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py:109]: +

**Review by akokhan — DISMISSED** [review]

File: platform/barefoot/sonic-platform-modules-bfn-newport/sonic_platform [hunk 4]
```diff
@@ -1 +0,0 @@
-../sonic-platform-modules-bfn-montara/sonic_platform
```
Comment (akokhan) [inline platform/barefoot/sonic-platform-modules-bfn-newport/sonic_platform:1]: please revert

**Comment by dmytroxIntel** [comment]
@lguohan , please take a look, thx

**Review by akokhan — APPROVED** [review]

**Comment by dmytroxIntel** [comment]
@qiluo-msft , @prgeor , @richardyu-ms, please approve and merge

**Review by prgeor — APPROVED** [review]

**Comment by dmytroxIntel** [comment]
@prgeor , please merge

**Comment by dmytroxIntel** [comment]
@qiluo-msft , @prgeor , @richardyu-ms, please merge

**Comment by akokhan** [comment]
@prgeor , @qiluo-msft , the PR is already approved. Please merge.

File: platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py [hunk 5]
```diff
@@ -75,14 +88,47 @@ def get_eeprom_path(self):
...
+        if not self.get_presence():
+            return None
+
+        if not EEPROM_CACHED_API_SUPPORT:
+            return super().read_eeprom(offset, num_bytes)
+
+
+        # TODO: remove when memory model is verified before requesting page that may not be supported
+        #
+        # If memory model is flat - the requested page can only be 00_LOWER or 00_UPPER,
```
Comment (prgeor) [inline platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py:107]: This is not correct, can you check [https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L31](https://github.com/sonic-net/sonic-platform-common/blob/master/sonic_platform_base/sonic_xcvr/api/public/cmis.py#L31)

Comment (dmytroxIntel) [reply platform/barefoot/sonic-platform-modules-bfn-montara/sonic_platform/sfp.py]: Thank you, removed workaround

**Comment by dmytroxIntel** [comment]
@prgeor , PR updated, comment fixed, please approve

**Pull request merged**